esp-hal/src/dma/mod.rs

```diff
@@ -1802,6 +1798,21 @@ pub trait Rx: crate::private::Sealed {
     fn waker(&self) -> &'static crate::asynch::AtomicWaker;
 }
 
+// NOTE(p4): because the P4 has two different GDMAs, we won't be able to use
+// `GenericPeripheralGuard`.
+cfg_if::cfg_if! {
+    if #[cfg(pdma)] {
+        type PeripheralGuard = system::GenericPeripheralGuard<{ system::Peripheral::Dma as u8}>;
```

Comment:
In the future this should be properly split up. No reason to enable all the DMAs upfront now.

Reply:
Yeah I guess, I ran into it with CryptoDMA and failed to connect the ideas.